scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -278,23 +278,40 @@ class Router(formatOps: FormatOps) {
           Split(Space, 0, ignoreIf = !canBeSpace),
           Split(afterCurlyNewlines, 1).withIndent(2, endOfFunction, Left)
         )
+
       case FormatToken(T.RightArrow(), right, _)
           if leftOwner.is[Term.Function] =>
-        val (endOfFunction, expiresOn) = functionExpire(
-          leftOwner.asInstanceOf[Term.Function]
-        )
-        val hasBlock =
-          nextNonComment(formatToken).right.isInstanceOf[T.LeftBrace]
-        val indent = // don't indent if the body is empty `{ x => }`
-          if (isEmptyFunctionBody(leftOwner) && !right.is[T.Comment]) 0
-          else 2
-        Seq(
-          Split(Space, 0, ignoreIf = isSingleLineComment(right))
-            .withPolicy(SingleLineBlock(endOfFunction)),
-          Split(Space, 0, ignoreIf = !hasBlock),
-          Split(Newline, 1 + nestedApplies(leftOwner), ignoreIf = hasBlock)
+        val lambda = leftOwner.asInstanceOf[Term.Function]
+        val (endOfFunction, expiresOn) = functionExpire(lambda)
+        val rightCommentedOut = isSingleLineComment(right)
+        val singleLineSplit =
+          Split(Space, 0, ignoreIf = rightCommentedOut)
+            .withPolicy(SingleLineBlock(endOfFunction))
+        def newlineSplit = {
+          val indent = // don't indent if the body is empty `{ x => }`
+            if (isEmptyFunctionBody(leftOwner) && !right.is[T.Comment]) 0
+            else 2
+          Split(Newline, 1 + nestedApplies(leftOwner))
             .withIndent(indent, endOfFunction, expiresOn)
-        )
+        }
+        val multiLineSplits =
+          if (rightCommentedOut) Seq(newlineSplit)
+          else if (!style.activeForEdition_2020_01)
+            Seq(
+              if (nextNonComment(formatToken).right.isInstanceOf[T.LeftBrace])
+                Split(Space, 0)
+              else newlineSplit
+            )
+          else if (right.is[T.LeftBrace] && (matching(right) eq endOfFunction))
+            Seq(
+              Split(Space, 0)
+                .withOptimalToken(getOptimalTokenFor(next(formatToken))),
+              newlineSplit
+            )
+          else
+            Seq(newlineSplit)
```

Comment:
If the right token of `=>` for lambda is a comment (`/** */` not a single line comment), we always add a newline like this change https://github.com/scalameta/scalafmt/pull/1612/files#diff-4ceca7814917e39c8477479c4cd26e35R143

Is this intended? I personally prefer not to add a new line there 🤔 

This change looks introduced with https://github.com/scalameta/scalafmt/pull/1612/commits/e8bb7141e5e3bb0da94fd0265604e453eb177699, could you elaborate the intension?